### PR TITLE
Add IMU compatible for KakuteH7Mini

### PR DESCRIPTION
**Pull request by jamming**
1)KakuteH7Mini_V1.1 have a MPU6000. Add MPU6000 driver for it.
2)KakuteH7Mini_V1.3 have a BMI270. According to the actual orientation of BMI270 on the PCB, ROTATION should be YAW270.
3)KakuteH7Mini_V1.5 have a ICM42688P. Add ICM42688P driver for it.
4) On other flight control software, Kakute H7Mini uses a 1058 BoardID. In order to make the bootloader universal, PX4 also changed the boardID to 1058.
5) Kakute H7Mini comes in both NAND FLASH and NOR FLASH versions. They all use the same MCU PIN-MAP. This target applies to both of them, so the board name has been changed to Kakute H7Mini.

### CODE REVIEW

**Comment by vincentpoont2** [comment]
We have tested this on all the hardware. 
@julianoes @dagar Can we merge this?

File: boards/holybro/kakuteh7mini/firmware.prototype [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
-    "board_id": 1054,
+    "board_id": 1058,
```
Comment (julianoes) [inline boards/holybro/kakuteh7mini/firmware.prototype:2]: It's a bit weird to "change" this like that.

What about existing boards? They suddenly have a different board ID between bootloader and firmware.

It's also confusing to have the same board with the same name twice in this list:
https://github.com/PX4/PX4-Bootloader/blob/e40999bfe9d6ea5da0dd717f5eb09adbd1f1c7d1/board_types.txt#L167C1-L171

Comment (julianoes) [reply boards/holybro/kakuteh7mini/firmware.prototype:2]: Did something change regarding NAND?
Should we add a new board ID?

**Comment by dagar** [comment]
I'm not sure what's going on with the board ID, do you want to bring in the IMU changes in the meantime or can we get this settled?

**Comment by julianoes** [comment]
Thanks @jamming. I've split the commit up into two pull requests for the two changes:
https://github.com/PX4/PX4-Autopilot/pull/23831
https://github.com/PX4/PX4-Autopilot/pull/23832

**Pull request closed without merging**